### PR TITLE
Fix "Warning: sprintf(): Too few arguments"

### PR DESCRIPTION
**Pull request by andrezrv**
WordPress 4.8 inserts the "read more" link in the update success message automatically. In this process, `sprintf()` receives only one replacement argument, so PHP throws a warning when two strings are expected to be replaced.

This is the commit in core where the modifications were made: https://github.com/WordPress/WordPress/commit/bab7eb143e3e3651cbb4165a9e904401dd2ed6de#diff-0e5737a4947b2884a63b3a50a59a9d3f.

This issue was already reported in #669, and a fix was proposed. However, I think that checking for the installed version of WordPress is better to keep backwards compatibility.

### CODE REVIEW

**Comment by GaryJones** [comment]
Quick review, but looks sensible. Thanks!

**Comment by jrfnl** [comment]
@andrezrv The travis error had nothing to do with your PR and has been fixed in the mean time. Would you mind rebasing the PR so we can get a passing build ? Thanks!
I'll test the PR after that and will merge it if I don't find any problems with it (I don't expect any issues).

**Comment by andrezrv** [comment]
@jrfnl Done, and I added one more commit to fix the end of line issue reported by Travis.

**Review by jrfnl — APPROVED** [review]
I've reproduced the issue on WP 4.8 and tested the fix on WP 4.6, 4.7 and 4.8 and found it all working as expected.

I made one small tweak to the code, replacing the `global` statement with using the locally cached version of the `$wp_version` variable for efficiency.
The variable is cached in the TGMPA constructor: https://github.com/TGMPA/TGM-Plugin-Activation/blob/develop/class-tgm-plugin-activation.php#L262

**Review by GaryJones — APPROVED** [review]

**Comment by Sharifur** [comment]
change this function to this 

https://gist.github.com/Sharifur/cdf71a63fb402eb8f38f6fbe03e0522b

**Comment by Ninetheme** [comment]
> change this function to this
> 
> https://gist.github.com/Sharifur/cdf71a63fb402eb8f38f6fbe03e0522b

thank you!

**Pull request merged**
